### ISSUE TITLE
predict_proba() with LinearSVC as classifier

### ISSUE THREAD

**Issue by graceleetr** [scikit-multilearn/scikit-multilearn#249]
Hi,  `predict_proba()` uses a `self.classifier.predict_proba()` but when a classifier is set to [LinearSVC,](https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html#sklearn.svm.LinearSVC) it does not have `predict_proba()`. it only has `decision_function()`. Is there any workaround?

**Comment by northern-64bit** [comment]
Hello @graceleetr !

You can wrap `LinearSVC` in a class that implements `predict_proba()` by calibrating the decision function outputs to probabilities. This involves using `CalibratedClassifierCV` from `scikit-learn`, which can calibrate the decision scores to probabilities. Here's how you can do it:

```python

from sklearn.svm import LinearSVC
from sklearn.calibration import CalibratedClassifierCV
from skmultilearn.problem_transform import BinaryRelevance
from sklearn.datasets import make_multilabel_classification
from sklearn.model_selection import train_test_split


X, y = make_multilabel_classification(n_samples=1000, n_features=20, n_classes=5, n_labels=2, random_state=42)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Initialize the LinearSVC classifier
base_clf = LinearSVC()

# Calibrate the classifier
calibrated_clf = CalibratedClassifierCV(base_clf)

# Use the calibrated classifier with for instance BinaryRelevance from scikit-multilearn
classifier = BinaryRelevance(classifier=calibrated_clf)

# Train the classifier
classifier.fit(X_train, y_train)

# Now you can use predict_proba
probabilities = classifier.predict_proba(X_test)

print(probabilities)
```
Note that the calibration may add some training time and complexity.

Also note that this may not always be normalized probabilities so this you may have to fix with:

```python

import numpy as np

# Convert the sparse matrix to a dense format for manipulation
prob_dense = probabilities.toarray()

# Normalize the probabilities row-wise
prob_normalized = prob_dense / np.sum(prob_dense, axis=1, keepdims=True)

# Check the sum again to ensure they are normalized
prob_sum_normalized = prob_normalized.sum(axis=1)

prob_normalized, prob_sum_normalized[:5]
```

I hope that this helps!
